fabric-object-builders/src/main/java/net/fabricmc/fabric/api/block/BlockSettingsExtensions.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.block;
+
+import net.fabricmc.fabric.mixin.builders.BlockSettingsHooks;
+import net.minecraft.block.Block.Settings;
+import net.minecraft.block.MaterialColor;
+import net.minecraft.item.Item;
+import net.minecraft.sound.BlockSoundGroup;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.Identifier;
+
+public final class BlockSettingsExtensions {
+    private BlockSettingsExtensions() {
+    }
+
+    public static void breakByHand(Settings settings, boolean breakByHand) {
+        FabricBlockSettings.computeExtraData(settings).breakByHand(breakByHand);
+    }
+
+    public static void breakByTool(Settings settings, Tag<Item> tag, int miningLevel) {
+        FabricBlockSettings.computeExtraData(settings).addMiningLevel(tag, miningLevel);
+    }
+
+    public static void hardness(Settings settings, float hardness) {
+        ((BlockSettingsHooks) settings).setHardness(hardness);
+    }
+
+    public static void resistance(Settings settings, float resistance) {
+        ((BlockSettingsHooks) settings).setResistance(Math.max(0.0F, resistance));
```

Comment:
Why is there a Math.max here? I understand that a resistance of < 0 might not make sense, but isn't there something further down that handles this?

Reply:
It matches what `Settings#strength` does, vanilla doesn't let you set the field without running through that so there's no further checking